cmd/thanos/tools_bucket.go

```diff
@@ -695,6 +695,7 @@ func registerBucketReplicate(app extkingpin.AppClause, objStoreConfig *extflag.P
 	maxTime := model.TimeOrDuration(cmd.Flag("max-time", "End of time range limit to replicate. Thanos Replicate will replicate only metrics, which happened earlier than this value. Option can be a constant time in RFC3339 format or time duration relative to current time, such as -1d or 2h45m. Valid duration units are ms, s, m, h, d, w, y.").
 		Default("9999-12-31T23:59:59Z"))
 	ids := cmd.Flag("id", "Block to be replicated to the destination bucket. IDs will be used to match blocks and other matchers will be ignored. When specified, this command will be run only once after successful replication. Repeated field").Strings()
+	ignoreMarkedForDeletion := cmd.Flag("ignore-marked-for-deletion", "Do not replicate blocks that have deletion mark.").Bool()
```

Comment:
I think this should even be the default TBH, surprised that it is not :smile:

Reply:
Well I was also a bit surprised  :D 
But didn't want to change the default behavior, but if we agree, I'd be happy to add `--replicated-marked-for-deletion` flag instead. Not sure if anyone could have a use case to replicated blocks that should be deleted.

Another thing is that it does not replicate the deletion marks (see https://github.com/thanos-io/thanos/pull/5118)

Reply:
Maybe let's keep the old behaviour for now :+1: